src/backend/distributed/commands/multi_copy.c

```diff
@@ -1953,6 +2229,9 @@ CitusCopyDestReceiverReceive(TupleTableSlot *slot, DestReceiver *dest)
 
 		/* find the partition column value */
 		partitionColumnValue = columnValues[partitionColumnIndex];
+
+		/* annoyingly this is evaluated twice, but at least we don't crash! */
```

Comment:
Also, just FYI: https://github.com/citusdata/citus/pull/1402#discussion_r119126686, in case you'd want to implement `RETURNING` 😄

Reply:
IIIInteresting. Maybe it's because adding the type casts does yet another walk of the tree? I'm actually surprised, the difference shouldn't be noticeable!